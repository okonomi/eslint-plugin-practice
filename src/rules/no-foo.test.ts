import { describe , it, expect } from "vitest"
import { createRuleTester } from "eslint-vitest-rule-tester"

import rule from "./no-foo"

describe("no-foo", () => {
  const { valid, invalid } = createRuleTester({
    name: 'no-foo',
    rule,
    configs: {
      // flat config options
      languageOptions: {
        parserOptions: {
          ecmaVersion: 2020,
          sourceType: 'module',
        },
      },
    }
  })

  it("should not report valid identifiers", async () => {
    const code = `const bar = 1;`
    const { result } = await valid({ code })

    expect(result.messages).toHaveLength(0)
  })

  it("should report 'foo' identifier", async () => {
    const code = `const foo = 1;`
    const errors = ['noFoo']
    const { result } = await invalid({ code, errors })

    expect(result.output).toMatchSnapshot()
  })
})
