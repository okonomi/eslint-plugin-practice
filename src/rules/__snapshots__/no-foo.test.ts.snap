// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`no-foo > should report 'foo' identifier 1`] = `"const foo = 1;"`;
